lib/Router.js

```diff
@@ -0,0 +1,198 @@
+	*/
+	router.post("/numbers", function (req, res) {
+		var clientNum = req.body.text;
+		var city = req.body.city;
+		var state = req.body.state;
+		self.searchNumbers(city, state).then(function (numbers) {
+			return self.orderNumber(numbers[0].number);
+		}).then(function (number) {
+			var alias = number.number;
+			conversations[alias] = [ clientNum, null, req.body.oneLiner ];
+			self.myNumbers.push(number);
+			self.getNumber(alias).update({
+				name : JSON.stringify(conversations[alias]), applicationId : applicationId }, function (err, result) {});
+			var firstMsg = "Thank you for choosing TextMe. Reply with your contact's number (ex. \"+15555555555\"). " +
+			"Send \"Get lost\" to erase this number";
+			return self.sendTextFromAlias(alias, clientNum, firstMsg);
+		}).then(function () {
+			res.send();//return 200ok so a timeout doesn't result in multiple requests
+		}).catch(function (err) {
+			res.status(500).send(err.message);
+		});
+	});
+
+	/*
+	*POSTs to /messages come from sms texts sent by either the client or the stranger.
+	*The text from the post message is parsed and sent to either the client or stranger
+	*depending on who sent it. If it is the first text in which the stranger number is unknown
+	*then stranger number is set to the text field of the sms message and the name field of the alias
+	*is updated to include the new stranger number
+	*/
+	router.post("/messages", function (req, res) {
+		var clientNum = conversations[req.body.to][0];
+		var strangerNum = conversations[req.body.to][1];
+		var text;
+		//this only checks if strangerNum is null and that the client isn't deleting the number.
+		//it would be better to also check if req.body.text is a valid number
+		if (!strangerNum && (req.body.text.toUpperCase() !== "GET LOST") && req.body.from === clientNum) {
+			if (req.body.text.match(e164Regex)){
+				strangerNum = req.body.text;
+				conversations[req.body.to][1] = strangerNum;
+				self.getNumber(req.body.to).update({
+					name : JSON.stringify(conversations[req.body.to]), applicationId : applicationId }, function (err, result) {});
+				text = conversations[req.body.to][2];
+				self.sendTextFromAlias(req.body.to, strangerNum, text);
+			}
+			else {
+				var unsuccessfulDelete = "TextMe: Invalid Number Format. Please send number in the form \"+15555555555\"";
+				self.sendTextFromAlias(req.body.to, clientNum, unsuccessfulDelete);
+			}
+		}
+		else {
+			text = req.body.text;
+			if (text.toUpperCase() === "GET LOST" && req.body.from === clientNum){
+				self.getNumber(req.body.to).delete(function (err, result) {
+					if (err) {
+						self.sendTextFromAlias(req.body.to, clientNum, "TextMe: Failed to delete");
+					}
+					else {
+						//Number to use for notifications from the app.
+						var successfulDelete = "TextMe: Successfully deleted alias. Thank you for choosing TextMe";
+						self.sendTextFromAlias(process.env.TEXTING_APP_NOTIFICATION_NUMBER, clientNum, successfulDelete);
```

Comment:
You should probably delete the number array from `myNumbers` here as well, to prevent it from growing unnecessarily.